homeassistant/components/config/config_entries.py

```diff
@@ -587,3 +639,61 @@ async def _async_matching_config_entries_json_fragments(
         )
         or (filter_is_not_helper and entry.domain not in integrations)
     ]
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        "type": "config_entries/subentries/list",
+        "entry_id": str,
+    }
+)
+@websocket_api.async_response
+async def config_subentry_list(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """List subentries of a config entry."""
+    entry = get_entry(hass, connection, msg["entry_id"], msg["id"])
+    if entry is None:
+        return
+
+    result = {
+        "subentries": {
```

Comment:
Any reason you prefer a dict? We don't have one for `config_entries_get`

Reply:
No reason. Frontend should decide if a dict or list is more convenient.

Reply:
Not sure, there are places where the first thing frontend does is convert the list to a dict 🤷